esmvaltool/cmorizers/data/formatters/datasets/esacci_aerosol.py

```diff
@@ -0,0 +1,282 @@
+        time_coord = iris.coords.DimCoord(
+            cf_units.date2num(timestamp, time_unit, time_calendar),
+            standard_name='time',
+            var_name='time',
+            units=cf_units.Unit(time_unit, calendar=time_calendar)
+        )
+        cube = iris.util.new_axis(cube)
+        cube.add_dim_coord(time_coord, 0)
+
+        for attr in drop_attrs:
+            if attr in cube.attributes.keys():
+                cube.attributes.pop(attr)
+
+        new_list.append(cube)
+
+    # make sure there is one cube for every day (daily data) or
+    # every month (monthly data) of the year
+    # (print debug info about missing days/months and add cube with
+    # nan to fill gaps
+
+    full_list = iris.cube.CubeList()
+    time_list = []
+
+    for cube in new_list:
+        loncoord = cube.coord('longitude')
+        latcoord = cube.coord('latitude')
+        loncoord.points = np.round(loncoord.core_points(), 3)
+        latcoord.points = np.round(latcoord.core_points(), 3)
+
+    # create list of available days/months ('time_list')
+
+    for cube in new_list:
+        timecoord = cube.coord('time')
+        cubetime = timecoord.units.num2date(timecoord.points)
+        time_list.append(cubetime)
+
+    # create cube list for every day/month of the year by adding
+    # cubes containing only nan to fill possible gaps
+
+    if is_daily:
+        loop_date = datetime(year0, 1, 1)
+        while loop_date <= datetime(year0, 12, 31):
+            date_available = False
+            for idx, cubetime in enumerate(time_list):
+                if loop_date == cubetime:
+                    date_available = True
+                    break
+            if date_available:
+                full_list.append(new_list[idx])
+            else:
+                logger.debug("No data available for %d", loop_date)
+                nan_cube = _create_nan_cube(new_list[0], loop_date.year,
+                                            loop_date.month, loop_date.day)
+                full_list.append(nan_cube)
+            loop_date += relativedelta.relativedelta(days=1)
+    else:
+        loop_date = datetime(year0, 1, 15)
+        print(loop_date)
+        while loop_date <= datetime(year0, 12, 31):
+            date_available = False
+            for idx, cubetime in enumerate(time_list):
+                if loop_date == cubetime:
+                    date_available = True
+                    break
+            if date_available:
+                full_list.append(new_list[idx])
+            else:
+                logger.debug("No data available for %d", loop_date)
+                nan_cube = _create_nan_cube(new_list[0], loop_date.year,
+                                            loop_date.month, loop_date.day)
+                full_list.append(nan_cube)
+            loop_date += relativedelta.relativedelta(months=1)
+
+    iris.util.unify_time_units(full_list)
+    cube = full_list.concatenate_cube()
```

Comment:
any chance this may fail (differing attributes etc)? Could try use the more accommodating `esmvalcore.preprocessor._io.concatenate` if it fails